src/Microsoft.ML.OnnxTransform/doc.xml

```diff
@@ -0,0 +1,92 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<doc>
+  <members>
+
+    <member name="OnnxTransform">
+      <summary>
+        Extracts hidden layers&apos; values from a pre-trained Onnx model.
+      </summary>
+      <remarks>
+        <para>
+          The OnnxTransform extracts the specified outputs from the operations computed on the graph (given the input(s)) using a pre-trained <a href="https://www.tensorflow.org">Onnx</a> model.
+          The transform takes as input the Tensorflow model together with the names of the inputs to the model and names of the operations for which output values will be extracted from the model.
+        </para>
+
+        <para>
+          This transform requires the <a href="https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.ML.TensorFlow/0.5.0-preview-26830-5">Microsoft.ML.TensorFlow</a> nuget to be installed.
+          The TensorflowTransform has following assumptions regarding the input, output and processing of data.
+        </para>
+        <list type="number">
+          <item>
+            <description>
+              The transform currently accepts the <a href="https://www.tensorflow.org/mobile/prepare_models">frozen TensorFlow model</a> file as input.
+            </description>
+          </item>
+          <item>
+            <description>The transform supports scoring only one example at a time.</description>
+          </item>
+          <item>
+            <description>The name of input column(s) should match the name of input(s) in Tensorflow model.</description>
+          </item>
+          <item>
+            <description>The name of each output column should match one of the operations in the Tensorflow graph.</description>
+          </item>
+          <item>
+            <description>Currently, float and double are the only acceptable data types for input/output.</description>
+          </item>
+          <item>
+            Upon success, the transform will introduce a new column in <see cref="IDataView"/> corresponding to each output column specified.
+          </item>
+        </list>
+
+        The inputs and outputs of a TensorFlow model can be obtained using the <a href="https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/graph_transforms/README.md#inspecting-graphs">
+          <code>summarize_graph</code> tool
+        </a>.
+
+      </remarks>
+      <example>
+        <code language="csharp">
+          pipeline.Add(new TextLoader(dataFile).CreateFrom&lt;MNISTData&gt;(useHeader: false));
+          pipeline.Add(new ColumnCopier(("NumericImageVec", "Input");
+          pipeline.Add(new TensorFlowScorer()
+          {
+            ModelFile = model_location;
+            InputColumns = new []{ "Input" };
+            OutputColumns = "Output"
+          }
```

Comment:
Make sure it's actually works.

Reply:
Added a snippet of unit test code, to ensure it works.

---
In reply to: [220397104](https://github.com/dotnet/machinelearning/pull/942#discussion_r220397104) [](ancestors = 220397104)